commons-hpcc/src/main/java/org/hpccsystems/commons/ecl/RecordDefinitionTranslator.java

```diff
@@ -416,13 +429,21 @@ else if (srcType == HpccSrcType.UTF16LE || srcType == HpccSrcType.UTF16BE)
                 String recordDefnName = "##" + hash + "##";
 
                 recordDefinitionMap.put(recordDefnName, definition);
-                return recordDefnName;
+                type = recordDefnName;
+                break;
             }
             default:
             {
                 throw new Exception("Unable to generate ECL unknown field type: " + field.getFieldType().description());
             }
         }
+
+        if (field.isBlob())
+        {
+            type += " {blob}";
```

Comment:
what typically precedes the ' {blob}' literal, and do we know for sure at this point type has been correctly populated with the expected pre-fix?